Optimized existing encryptions + added additional ones

KILI-BC opened the pull request:
- Optimized code of existing encryptions
- Retested Code of existing encryptions
- Added + implemented + linked Rot13, Rot47 and Atbash-encryptions
- Tested additional encryptions

KILI-BC committed:
Optimized code of encrypt.c

- Added function roll_in_alphabeth(int i, int shift, int alphabeth_size)
- Simplified more complicated lines of code with use of tolower-function
- tested functions

KILI-BC committed:
Added Rot13-Encryption

- added name in encrypttype-enum
- implemented + tested encrypt-function
- linked functions in encrypt_init

KILI-BC committed:
Added Rot47-Encryption

- added name in encrypttype
- implemented + tested encrypt function
- linked necessary functions

KILI-BC committed:
Added Atbash-Encryption

- added name in encrypttype
- implemented + tested encrypt-function
- linked functions in encrypt_init

b0iizz commented on src/encrypt.c:
```diff
@@ -11,6 +11,8 @@ static void encrypt_none_state_free(void *state);
 static void encrypt_none_encode(char **str, void *key, void *state);
 static void encrypt_none_decode(char **code, void *key, void *state);
 
+static int roll_in_alphabeth(int i, int shift, int alphabeth_size);
```
```suggestion
static int roll_in_alphabet(int i, int shift, int alphabet_size);
```
Das ist die Korrekte Schreibweise (auch auf Englisch)

b0iizz commented on src/encrypt.c:
```diff
@@ -38,12 +46,46 @@ void encrypt_init()
+    encryptors[ENCRYPT_ATBASH].decode = &encrypt_atbash_encode;
+}
+
+static int roll_in_alphabeth(int i, int shift, int alphabeth_size)
```
s.o.

b0iizz commented on src/encrypt.c:
```diff
@@ -199,17 +237,68 @@ static void encrypt_vigenere_decode(char **codetext, void *key, void *state)
+    }
+    (void)key;
+    (void)state;
+}
```
Am Ende bitte eine Leerzeile lassen

KILI-BC committed:
Update src/encrypt.c

Corrected spelling of alphabet

Co-authored-by: B0IIZZ <general.boiizz@gmail.com>

KILI-BC committed:
Corrected Spelling mistakes

- replaced "alphabeth" with "alphabet"
- added newline at the end of the file

b0iizz committed:
Fix remaining spelling mistakes

KILI-BC committed:
Added substitution cipher

- added ENCRYPT_SUBSTITUTION to enum
- added +linked + implemented + tested necessary functions

KILI-BC committed:
Fixed allocation-errors in key_parse-functions

- Corrected formatting
- Added checking after memory-allocation

KILI-BC committed:
Fixed malloc-errors

- added check after memory allocation

KILI-BC committed:
fixed errors + added pairwise substitution

- fixed error in decryption of substitution cipher
-added ENCRYPT_PAIRWISE_SUBSTITUTION to enum
- implemented, tested + linked pairwise_substitution_key_parse

KILI-BC committed:
Merged online changes with local changes

- fixed discrepancies between local version and online version

b0iizz committed:
Merge branch 'master' into master
